File: HomeworkDay08/app/src/main/java/com/example/homework_day_08/MainActivity.java
package com.example.homework_day_08;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.example.homework_day_08.fragment.TestFragment;
import com.example.homework_day_08.tools.TestKotlin;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainFragment";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        addFragmentManager();
        setButtonOnClickListener();
        TestKotlin testKotlin = new TestKotlin();
        Log.d(TAG, String.valueOf(testKotlin.sum(1,2)));
    }


    private void addFragmentManager() {
        getSupportFragmentManager().beginTransaction()
                .setReorderingAllowed(true)
                .add(R.id.test_fragment_container_view, TestFragment.class, null)
                .commit();
    }

    private void setButtonOnClickListener() {
        Button button;
        button = findViewById(R.id.to_test_activity);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, HomeworkActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: OverallReview/app/src/main/java/com/spcrey/overall_review/MainActivity.kt
package com.spcrey.overall_review

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.spcrey.overall_review.fragment.TestFragment

class MainActivity : AppCompatActivity() {
    companion object {
        private const val TAG = "MainActivity"
    }

    private val btnToTestActivity by lazy {
        findViewById<Button>(R.id.btn_to_test_activity)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportFragmentManager.beginTransaction()
            .setReorderingAllowed(true)
            .add(
                R.id.fragment_container_view,
                TestFragment::class.java,
                null,
                "TAG_FRAGMENT"
            )
            .commit()

        btnToTestActivity.setOnClickListener {
            val intent = Intent(this, TestActivity::class.java)
            startActivity(intent)
        }
    }
}
